timely_beliefs/beliefs/classes.py

```diff
@@ -1467,6 +1492,124 @@ def resample_events(
+        :param inplace:           If True, the new beliefs are added to the BeliefsDataFrame in place.
+                                  If False, the new beliefs are returned (use pd.concat to add them yourself).
+        """
+        if forecaster is None:
+            forecaster = NaiveForecaster(strategy="last")
+        if event_start is not None:
+            if event_time_window != (None, None):
+                raise ValueError(
+                    "Cannot pass both an event start and event start window."
+                )
+            event_time_window = (event_start, event_start + self.event_resolution)
+
+        # Check assumptions
+        if self.sensor.knowledge_time(event_time_window[0]) < belief_time:
+            # No backcasting
+            raise NotImplementedError("Backcasting is not implemented.")
+        if self.empty:
+            # At least 1 belief
+            raise NotImplementedError(
+                "Cannot form new beliefs given an empty data frame."
+            )
+        if self.lineage.number_of_sources != 1:
+            # Single source only
+            raise NotImplementedError(
+                "Cannot form new beliefs given multi-sourced data. Consider picking 1 source, e.g. using: df = df[df.index.get_level_values('source') == df.lineage.sources[0]]"
+            )
+        if self.lineage.probabilistic_depth != 1:
+            # Deterministic beliefs only
+            raise NotImplementedError(
+                "Cannot form new beliefs given probabilistic data. Consider pre-computing deterministic beliefs, e.g. using: df = df.make_deterministic()"
+            )
+        if self.lineage.number_of_beliefs != self.lineage.number_of_events:
+            # One belief per event
+            raise NotImplementedError(
+                "Cannot form new beliefs given multiple beliefs about the same event. Consider picking a single belief per event, e.g. using: df = belief_utils.select_most_recent_belief(df)"
+            )
+
+        df = self
+
+        # Do NOT use future data to predict NOR to fit
+        if "belief_time" not in df.index.names:
+            df = df.convert_index_from_belief_horizon_to_time()
+        df = df[df.index.get_level_values("belief_time") < belief_time]
+
+        # Simplify index
+        df = df.reset_index()[["event_start", "event_value"]].set_index("event_start")
+
+        # Convert to local time in UTC, which sktime expects
+        tz = event_time_window[0].tzinfo
+        utc = pytz.utc
+        event_time_window = tuple(
+            dt.astimezone(utc).replace(tzinfo=None) for dt in event_time_window
+        )
+        df.index = df.index.tz_convert(utc).tz_localize(None)
+
+        # Resample to the resolution the data already has, just to set the index frequency, which sktime expects
+        df = df.reindex(
+            pd.date_range(
+                df.index[0],
+                event_time_window[0],
```

Comment:
I expected event_time_window[-1] here. Or maybe belief_time. Can you motivate this?

Reply:
I'm adding this motivation:

The new index ends just before the start of the first forecast (which may introduce trailing NaN values), as sktime expects no gap between the indices of the input and forecasts when applying seasonal periodicity.